test/unit/dao/FeiDao.test.ts

```diff
@@ -0,0 +1,189 @@
+import { expectRevert, time, getCore, getAddresses } from '../../helpers';
+import { expect } from 'chai';
+import hre, { artifacts, ethers, network } from 'hardhat';
+import { Signer } from 'ethers';
+import { TransactionResponse } from '@ethersproject/providers';
+
+const FeiDAO = artifacts.readArtifactSync('FeiDAO');
+const Timelock = artifacts.readArtifactSync('Timelock');
+const Tribe = artifacts.readArtifactSync('Tribe');
+
+const toBN = ethers.BigNumber.from;
+
+describe('FeiDAO', function () {
+  let userAddress: string;
+
+  const impersonatedSigners: { [key: string]: Signer } = {};
+
+  before(async () => {
+    const addresses = await getAddresses();
+
+    // add any addresses you want to impersonate here
+    const impersonatedAddresses = [addresses.userAddress];
+
+    for (const address of impersonatedAddresses) {
+      await hre.network.provider.request({
+        method: 'hardhat_impersonateAccount',
+        params: [address]
+      });
+
+      impersonatedSigners[address] = await ethers.getSigner(address);
+    }
+  });
+
+  beforeEach(async function () {
+    ({ userAddress } = await getAddresses());
+
+    await network.provider.request({
+      method: 'hardhat_reset',
+      params: []
+    });
+
+    this.core = await getCore();
+    const tribeAddress = await this.core.tribe();
+
+    // Deploy timelock with vanilla user admin
+    const timelockDeployer = await ethers.getContractFactory(Timelock.abi, Timelock.bytecode);
+    this.timelock = await timelockDeployer.deploy(userAddress, 1, 1); // delay and min delay 1s
+
+    // Deploy new Fei DAO
+    const feiDAODeployer = await ethers.getContractFactory(FeiDAO.abi, FeiDAO.bytecode);
+    this.feiDAO = await feiDAODeployer.deploy(tribeAddress, this.timelock.address);
+
+    // Update timelock admin to Fei DAO
+    await hre.network.provider.request({
```

Comment:
Not something you need to fix, but I'm going to add a utility method that does this requesting via hardhat and such for us, so that getting a new signer is a one-liner.

Reply:
sick